tools/api_proto_breaking_change_detector/detector_test.py

```diff
@@ -0,0 +1,103 @@
+from pathlib import Path
+import unittest
+
+from detector import BufWrapper, ChangeDetectorInitializeError
+
+
+class BufTests(unittest.TestCase):
+
+    def run_buf_test(self, testname, is_breaking, expects_changes, additional_args=None):
+        tests_path = Path(
+            Path(__file__).absolute().parent.parent, "testdata",
+            "api_proto_breaking_change_detector", "breaking" if is_breaking else "allowed")
+
+        current = Path(tests_path, f"{testname}_current")
+        changed = Path(tests_path, f"{testname}_next")
+
+        breaking_response = BufWrapper.is_breaking(current, changed, additional_args)
+        self.assertEqual(breaking_response, is_breaking)
+
+        lock_file_changed_response = BufWrapper.lock_file_changed(current, changed, additional_args)
+        self.assertEqual(lock_file_changed_response, expects_changes)
+
+
+class TestBreakingChanges(BufTests):
```

Comment:
From a design point-of-view, note that these tests are not necessarily a type of `BufTests` (e.g., could be `ProtolockTests`). As such, separating the executor (`buf` or something else) from the tests may be better.

Reply:
pushed a commit hopefully organizes tests a bit better